oslc_prolog/lib/oslc_dispatch.pl

```diff
@@ -69,15 +75,22 @@
   _{ iri_spec: Prefix:ResourceSegments,
        method: Method } :< Context,
   atomic_list_concat(ResourceSegments, '/', Resource),
-  findall(handler(Method, IRI, Handler, Priority), (
+  findall(handler(Handler, Priority), (
     handler(Method, ISPrefix:ISResource, Handler, Priority),
-    match_wildcard(ISPrefix, Prefix, ISResource, Resource),
-    rdf_global_id(Prefix:Resource, IRI)
+    match_wildcard(ISPrefix, Prefix, ISResource, Resource)
   ), Handlers),
-  predsort(handler_compare, Handlers, [handler(Method, IRI, Module:Predicate, _)|_]),
+  predsort(handler_compare, Handlers, SortedHandlers),
+  rdf_global_id(Prefix:Resource, IRI),
   NewContext = Context.put(iri, IRI).put(iri_spec, Prefix:Resource),
-  T =.. [Predicate, NewContext],
-  call(Module:T).
+  dispatch_to_handlers(NewContext, SortedHandlers).
+
+dispatch_to_handlers(Context, [handler(Module:Predicate, _)|T]) :-
```

Comment:
FEATURE: if a resource handler returns `false`, dispatcher will call the next handler with lower priority that matches the resource; if no handlers left - OSLC Core will handle it.